spring-xd-dirt/src/main/java/org/springframework/xd/dirt/rest/StreamsController.java

```diff
@@ -67,4 +75,27 @@ public StreamsController(StreamDeployer streamDeployer, StreamDefinitionReposito
 	protected StreamDefinition createDefinition(String name, String definition) {
 		return new StreamDefinition(name, definition);
 	}
+
+	@ResponseBody
+	@RequestMapping(value = "/clean/{transport}/{stream}")
```

Comment:
I guess `DELETE` is the best choice for `clean`.
But does `curl` support such a HTTP method ?